### ISSUE TITLE
Error while using the bin() function

### ISSUE THREAD

**Issue by LucasSauniere** [lightkurve/lightkurve#941]
#### Problem description

When I'm using the bin function, I have an error, but I can't find where it's coming from.
I've tried changing binsize into bins but I still have the same error.
The other functions like flatten and fold are working so I'm wondering what's wrong in my code.
I even tried to upgrade lightkurve via pip but it's still not working.

#### Example

TypeError                                 Traceback (most recent call last)
<ipython-input-19-e184bb836f5a> in <module>
      1 #lc.remove_nans().flatten(window_length=401).fold(period=3.5225).plot();
----> 2 lc.remove_nans().flatten(window_length=401).fold(period=3.5225).bin(bins=10).plot();

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\lightkurve\lightcurve.py in bin(self, binsize, bins, method)
   1034                 warnings.simplefilter("ignore", RuntimeWarning)
   1035                 binned_stat = binned_statistic(bin_by_array, values_to_bin,
-> 1036                     statistic=bin_function, bins=bin_edges).statistic
   1037                 setattr(binned_lc, attr, binned_stat)
   1038 

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\scipy\stats\_binned_statistic.py in binned_statistic(x, values, statistic, bins, range)
    181 
    182     medians, edges, binnumbers = binned_statistic_dd(
--> 183         [x], values, statistic, bins, range)
    184 
    185     return BinnedStatisticResult(medians, edges[0], binnumbers)

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\scipy\stats\_binned_statistic.py in binned_statistic_dd(sample, values, statistic, bins, range, expand_binnumbers, binned_statistic_result)
    614         result.fill(null)
    615         _calc_binned_statistic(Vdim, binnumbers, result, values, statistic,
--> 616                                is_callable=True)
    617 
    618     # Shape into a proper matrix

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\scipy\stats\_binned_statistic.py in _calc_binned_statistic(Vdim, bin_numbers, result, values, stat_func, is_callable)
    646             # or more for speed up.
    647             if is_callable or not (stat_func is np.std and len(bin_map[i]) < 2):
--> 648                 result[vv, i] = stat_func(bin_map[i])
    649 
    650 

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\lightkurve\lightcurve.py in <lambda>(x)
    988         # Assume the errors combine as the root-mean-square
    989         rmse_func = lambda x: np.sqrt(np.nansum(x**2))/len(x) \
--> 990                               if np.any(np.isfinite(x)) else np.nan
    991         statistic_mapper = {'flux': method_func,
    992                             'time': method_func,

TypeError: unsupported operand type(s) for ** or pow(): 'list' and 'int'
```


#### Environment

-  platform (e.g. Linux, OSX, Windows): Windows 10
-  lightkurve version (e.g. 1.0b6): Version: 1.11.3
-  installation method (e.g. pip, conda, source): pip

**Comment by barentsen** [comment]
@LucasSauniere I'm sorry to hear you are experiencing this bug.  Thank you for taking the time to open an issue!

Would you be willing to post a complete example that we can issue to reproduce the issue?  i.e. can you help me understand how to obtain a light curve `lc` in which the issue appears?  Thank you!

**Comment by LucasSauniere** [comment]
@barentsen thank you for your quick answer ! 

I've been thinking that maybe (because it's the only function compared to fold() and flatten() that's not working) a module is missing on my laptop even though I executed the cmd to install all of the needed ones for lightkurve.

Please find the beginning of my program below, built on an example of the tutorial.

###
print("Hello Wonderful Person")

from lightkurve import search_targetpixelfile
pixelfile = search_targetpixelfile("KIC 8462852", quarter=16).download(quality_bitmask='default');

pixelfile.plot(frame=1000);

lc = pixelfile.to_lightcurve(aperture_mask='all');

lc.time, lc.flux

lc.plot();

pixelFile = search_targetpixelfile('KIC 6922244', quarter=4).download()

lc = pixelFile.to_lightcurve(aperture_mask=pixelFile.pipeline_mask)

lc.plot();

flat_lc = lc.flatten(window_length=401)
flat_lc.plot();

folded_lc = flat_lc.fold(period=3.522)
folded_lc.plot();

binned_lc = folded_lc.bin(binsize=10)
binned_lc.plot();
###

**Comment by LucasSauniere** [comment]
Hello, I didn't get any news from you @barentsen. Did you find any solution for my problem ? 

Thank you by advance, and sorry to bother you.

Lucas

**Comment by barentsen** [comment]
@LucasSauniere Apologies for the delay, I haven't had to chance to look into the issue yet, but hope to do so tomorrow!

I should mention that Lightkurve v2.0.0 has been released today, which includes significant modifications to the `bin()` feature.  Regardless, I also do intend to fix the issue in v1.11.3 if it turns out to be a bug.

Stay tuned!

**Comment by barentsen** [comment]
@LucasSauniere Good news: the error does not occur in Lightkurve v2.0 (released earlier this week). This is not surprising because the `bin()` function was completely re-implemented in v2.0.  Example:

<img width="901" alt="Screen Shot 2021-02-17 at 7 59 24 PM" src="https://user-images.githubusercontent.com/817669/108303380-ab4ede00-715a-11eb-9883-882b38cf893b.png">

**Comment by barentsen** [comment]
I must add that I was able to reproduce the bug in Lightkurve v1.11.3.  I could look into fixing it in v1.11.3 and creating a new bugfix release (i.e. v1.11.4) if there is a specific need or significant demand.

My preference would be for users to upgrade to v2.0 if possible.  This can be done using `python -m pip install lightkurve --upgrade`.

I hope this helps.  Feel free to re-open this issue if you have any further questions or comments!

**Comment by LucasSauniere** [comment]
Thank you so much for your help @barentsen ! 
I tried to upgrade LightKurve to v2.0 in the past days but unfortunately the upgrade didn't work, and now I repeated the cmd to upgrade it and it's working fine, I have the v2 installed.

I'm really enjoying LightKurve and can't wait to see it's new features ! Thank you a lot again :)

**Comment by LucasSauniere** [comment]
Hello @barentsen and sorry to bother you again..

I have upgraded LightKurve and now have version 2.0.1 but the bin function is still not working.
I followed the instructions written one the LightCurve tutorial, and my code is the same one as the website:

#####

from lightkurve import search_targetpixelfile
tpf = search_targetpixelfile('KIC 6922244',author="Kepler",cadence="long",quarter=4).download()
lc = tpf.to_lightcurve(aperture_mask=tpf.pipeline_mask)

lc.plot();

flat_lc = lc.flatten(window_length=401)
flat_lc.plot();

folded_lc = flat_lc.fold(period=3.5225)
folded_lc.plot();

binned_lc = folded_lc.bin(time_bin_size=0.01)
binned_lc.plot()

#####

And here comes the error : 

ValueError: Input values did not match the format class jd:
TypeError: for jd class, input should be doubles, string, or Decimal, and second values are only allowed for doubles.

The full error is written at the end of this post.

Thank you in advance for your help.


######

TypeError                                 Traceback (most recent call last)
c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\astropy\time\core.py in _get_time_fmt(self, val, val2, format, scale, precision, in_subfmt, out_subfmt)
    429             try:
--> 430                 return cls(val, val2, scale, precision, in_subfmt, out_subfmt)
    431             except UnitConversionError:

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\astropy\time\formats.py in __init__(self, val1, val2, scale, precision, in_subfmt, out_subfmt, from_jd)
    148         else:
--> 149             val1, val2 = self._check_val_type(val1, val2)
    150             self.set_jds(val1, val2)

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\astropy\time\formats.py in _check_val_type(self, val1, val2)
    432                 'and second values are only allowed for doubles.'
--> 433                 .format(self.name))
    434 

TypeError: for jd class, input should be doubles, string, or Decimal, and second values are only allowed for doubles.

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
<ipython-input-14-06840401b516> in <module>
----> 1 binned_lc = folded_lc.bin(time_bin_size=0.01)
      2 p=binned_lc.plot()
      3 from matplotlib import pyplot as plt
      4 plt.show(p)

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\astropy\utils\decorators.py in wrapper(*args, **kwargs)
    533                     warnings.warn(message, warning_type, stacklevel=2)
    534 
--> 535             return function(*args, **kwargs)
    536 
    537         return wrapper

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\lightkurve\lightcurve.py in bin(self, time_bin_size, time_bin_start, n_bins, aggregate_func, binsize)
   1281         if not isinstance(time_bin_start, Time):
   1282             time_bin_start = Time(
-> 1283                 time_bin_start, format=self.time.format, scale=self.time.scale
   1284             )
   1285 

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\astropy\time\core.py in __init__(self, val, val2, format, scale, precision, in_subfmt, out_subfmt, location, copy)
   1525         else:
   1526             self._init_from_vals(val, val2, format, scale, copy,
-> 1527                                  precision, in_subfmt, out_subfmt)
   1528             self.SCALES = TIME_TYPES[self.scale]
   1529 

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\astropy\time\core.py in _init_from_vals(self, val, val2, format, scale, copy, precision, in_subfmt, out_subfmt)
    371         # Parse / convert input values into internal jd1, jd2 based on format
    372         self._time = self._get_time_fmt(val, val2, format, scale,
--> 373                                         precision, in_subfmt, out_subfmt)
    374         self._format = self._time.name
    375 

c:\users\lucas\appdata\local\programs\python\python37\lib\site-packages\astropy\time\core.py in _get_time_fmt(self, val, val2, format, scale, precision, in_subfmt, out_subfmt)
    440                         + os.linesep
    441                         + f'{err.__class__.__name__}: {err}'
--> 442                     ) from err
    443                 else:
    444                     problems[name] = err

**Comment by barentsen** [comment]
Thanks for reporting this problem @LucasSauniere!

This is a different bug.  Fortunately, it was fixed last night (#953).  The fix will be included in Lightkurve v2.0.2, which I expect to release before the start of the weekend.  I will confirm here.

**Comment by LucasSauniere** [comment]
Thanks a lot @barentsen ! I can't wait to have the update in order to fix the issue, by the way the version 2.0.1 already has some amazing features :)

**Comment by barentsen** [comment]
@LucasSauniere Lightkurve v2.0.2 has now been released on pypi.  It includes the bugfix for this issue.

You can upgrade using `python -m pip install lightkurve --upgrade`.

Please try again, and feel welcome to open new issues if you encounter problems.  We appreciate all the feedback and testing we can get! :+1:
